test/module_definition/parameter_documentation_test.py

```diff
@@ -34,3 +37,112 @@ def test_illegal_formatting_fails(text):
     except ValueError:
         return
     assert False
+
+
+def test_duplicate_in_out_attributions_are_not_allowed():
+    # given:
+    text = '@param[in, out] test'
+
+    # when:
+    try:
+        ParameterDocumentation.from_param_string(text)
+    except ValueError:
+        return
+    assert False
+
+
+@pytest.mark.parametrize('text,fixed_length', [
+    ('@param[in, fixed-length=3] test', 3),
+    ('@param[in, fixed-length=2] test', 2),
+    ('@param[in,fixed-length=5] test', 5),
+    ('@param[in,fixed-length =8] test', 8),
+    ('@param[in,fixed-length= 9] test', 9),
+])
+def test_parse_fixed_length(text, fixed_length):
+    # when:
+    parameter_documentation = ParameterDocumentation.from_param_string(text)
+
+    # then:
+    assert parameter_documentation.identifier == 'test'
+    assert parameter_documentation.kind == ParameterKind.kind_in()
+    assert parameter_documentation.fixed_length == fixed_length
+
+
+@pytest.mark.parametrize('text', [
+    '@param[in, fixed-length] test',
+    '@param[in, fixed-length=4, fixed-length=5] test'
+])
+def test_illegal_fixed_length(text):
```

Comment:
These tests also test `ActiveAttributions`. Additionally, the second one is a generic test which checks that no attribution is allowed to appear twice. Make that clear and adapt the tests.